pydicom/encoders/base.py

```diff
@@ -290,22 +297,28 @@ def _encode_dataset(
             )
 
         # Pixel Data is compressed
-        # Note that from this point on we require at least numpy be available
-        if decoding_plugin:
-            ds.convert_pixel_data(handler_name=decoding_plugin)
-
-        arr = ds.pixel_array
-
-        if kwargs['number_of_frames'] > 1 or len(arr.shape) == 4:
-            if idx is None:
-                raise ValueError(
-                    "The frame 'idx' is required for multi-frame pixel data"
-                )
-
-            arr = arr[idx]
+        raise ValueError(
+            "The dataset must be decompressed and correct 'Transfer "
+            "Syntax UID' and 'Photometric Interpretation' values set before "
+            "attempting to compress"
+        )
 
-        src = self._preprocess(arr, **kwargs)
-        return self._process(src, encoding_plugin, **kwargs)
+        # Note that from this point on we require at least numpy be available
+        # if decoding_plugin:
+        #     ds.convert_pixel_data(handler_name=decoding_plugin)
+        #
+        # arr = ds.pixel_array
+        #
+        # if kwargs['number_of_frames'] > 1 or len(arr.shape) == 4:
+        #     if idx is None:
+        #         raise ValueError(
+        #             "The frame 'idx' is required for multi-frame pixel data"
+        #         )
+        #
+        #     arr = arr[idx]
+        #
+        # src = self._preprocess(arr, **kwargs)
+        # return self._process(src, encoding_plugin, **kwargs)
```

Comment:
Are you planning to bring the implicit decompression back in?
EDIT: Ok, I have seen the xfailed tests, so the answer is obvioulsy "yes".

Reply:
Yes, this is solvable for some compression schemes like RLE Lossless and J2K (because I can parse the codestream and infer the actual colour space of the raw data), but probably not the original JPEG.